docker/docker.go

```diff
@@ -58,6 +58,10 @@ func StartContainerWithRetry(l zerolog.Logger, req tc.GenericContainerRequest) (
 				return nil, err
 			}
 		}
+		// TODO we should dynamically check if container was removed, otherwise retry will fail anyway, just becasuse
```

Comment:
I remember this was introduced because of Geth ports mapping issue, can we check if we can remove it now @AnieeG ?

Reply:
IMHO if we could remove it would be sweet, because I am yet to see a container that fails to start the first time, but then succeeds.

Reply:
Just chiming in here that I have seen this be helpful in CI in multiple cases, usually around the docker pull failing one of the times because of a server hiccup but then correctly pulls it the second or third time.

Reply:
Although we could add our own pull logic with retries so testcontainers-go doesn't have to do it.

Reply:
I agree with @tateexon here about custom pulling logic instead of retrying whole `tc.GenericContainer()`. Here's how testcontainers-go [does it](https://github.com/testcontainers/testcontainers-go/blob/main/docker.go#L1175). @Tofel but, the pulling logic with retry can be a separate ticket